mypy/semanal.py

```diff
@@ -1826,6 +1826,10 @@ def analyze_lvalue(self, lval: Lvalue, nested: bool = False,
                 lval.kind = LDEF
                 lval.fullname = lval.name
                 self.add_local(v, lval)
+                if isinstance(lval, NameExpr) and lval.name == '_' and self.is_func_scope():
```

Comment:
First `isinstance()` should be unnecessary, since we are already in a huge `if isinstance(lval, NameExpr)` block. Also the last check `self.is_func_scope()` is not needed, because it is equivalent to `self.locals[-1] is not None` where we are now. But it is up to you.